tests/agent_server/test_agent_code_loading.py

```diff
@@ -318,7 +329,24 @@ def test():
         version=version,
         resource_types=["test::Test"],
     )
-    await agent.ensure_code(install_spec)
-
-    assert agent._env.are_installed(["pkg", "dep-a"])
-    assert not agent._env.are_installed(["dep-b", "dep-c"])
+    executor = await agent.executor_manager.get_executor("agent1", "localhost", install_spec)
+
+    installed_packages = executor.executor_virtual_env.get_installed_packages()
+
+    def check_packages(
```

Comment:
Same question here, why is this change required?

Reply:
I'm not sure, this ties into my comment on the file https://github.com/inmanta/inmanta-core/pull/7867#discussion_r1691485611
As I understand it:
- `test_agent_code_loading` is aimed at testing executors in `executor-mode==threaded`,  because it is testing agent code reloading (and incidentally using in-process specific code e.g. `agent.executor_manager.ensure_code()`
- `test_agent_installs_dependency_containing_extras` (in its state prior to this PR) was only testing executors in `executor-mode==forking` (the default mode) and using forking mode specific code e.g. `executor.executor_virtual_env`

So I adapted the tests to each run against a specific executor mode. At the root of this is the fact that `ensure_code()` was moved from the `Agent` to the `InProcessExecutorManager` and I'm not sure how to adapt the tests that were relying on this.

Reply:
> I adapted the tests to each run against a specific executor mode

This is the part I don't get. You say that they were already running against a specific executor mode, so what does this change exactly? Is it that previously it didn't use the executor so we didn't care if it picked the wrong one for the mode we were testing?

Reply:
That would answer the question for the `server_pre_start` change, but why did these assertions need to be changed? `executor.executor_virtual_env` is the same as `agent._env` was, isn't it?

Reply:
> Is it that previously it didn't use the executor so we didn't care if it picked the wrong one for the mode we were testing?

- For the `test_agent_code_loading` I think so, yes. Since it is testing `ensure_code()` which was moved from the `Agent` to the `InProcessExecutorManager` I think this test is only relevant for `threaded` mode.

> executor.executor_virtual_env is the same as agent._env was, isn't it?

Not exactly, the former is a https://github.com/inmanta/inmanta-core/blob/cfdd76f035e218dda683c34cd3d26d37b130c825/src/inmanta/agent/executor.py#L247 whereas the latter is a 
https://github.com/inmanta/inmanta-core/blob/82842dfb76b1fb120576b884723758fcaf066e5d/src/inmanta/env.py#L1195

and so it doesn't have a `are_installed()` method.

Reply:
Right, it's not activated.